### PR TITLE
use the mongo client exposed via fhdb

### PR DESCRIPTION
**Pull request by danbev**

### CODE REVIEW

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* BLOCKER [Add a semicolon at the end of this statement.](https://github.com/feedhenry/fh-mbaas-api/blob/5d98a27f42a66549a9a8eb787336f96f4f00f2ea/lib/sync-UpdatesModel_mongo.js#L25) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by danbev** [comment]
@david-martin Would you mind taking a look at this? This is my attempt to use the new functionality that @aidenkeating added which exposes the mongo client.

File: lib/sync-UpdatesModel_mongo.js [hunk 1]
```diff
@@ -1,19 +1,9 @@
...
-};
-
-var setConnectionUrl = function(url) {
-  MongoClient.connect(url, function(err, db) {
-    if (err) throw err;
-    syncUtil.doLog(syncUtil.SYNC_LOGGER, 'info', 'sync-UpdatesModel MongoClient connected');
-
-    mongo = db;
-  });
+  mongo = fhdb.getMongoClient();
```
Comment (david-martin) [inline lib/sync-UpdatesModel_mongo.js]: This is giving an error
```js
/home/dmartin/work/fh-mbaas-api/lib/sync-UpdatesModel_mongo.js:6
  mongo = fhdb.getMongoClient();
               ^

TypeError: fhdb.getMongoClient is not a function
```
I think the function may not be exported completed through fh-db and to `db.js` in fh-mbaas-api.
i'll take a look into it.
I'm using master fh-db, npm linked into this fh-mbaas-api.

Comment (danbev) [inline lib/sync-UpdatesModel_mongo.js]: Ah sorry about that. I linked it locally and forgot about it :(

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* BLOCKER [Add a semicolon at the end of this statement.](https://github.com/feedhenry/fh-mbaas-api/blob/9bbb66f9aa4c5c02c45143e18fb6a5dcc3999588/lib/sync-UpdatesModel_mongo.js#L25) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by danbev** [comment]
I've triggered a [rebuild](http://bob.feedhenry.net:8080/job/fh-mbaas-api_pullrequests/510/) after publishing `fh-db@1.4.0` to `http://npm.skunkhenry.com`.

**Comment by danbev** [comment]
@david-martin You mentioned yesterday that you were having issue running the sync server locally. I've tried running [sync-cloud](https://github.com/feedhenry-templates/sync-cloud) and [sync-cordova-app](https://github.com/feedhenry-templates/sync-cordova-app) without seeing any errors. 

Do you have any stacktrace that I can take a look at to investigate further?

**Comment by david-martin** [comment]
TLDR: its still using the fhdb DataSet model, even though we have mongdb running locally.

@danbev when running the sync server locally with `grunt serve`, the default config for mongodb (down in fh-db) is 127.0.0.1:6379 https://github.com/feedhenry/fh-db/blob/master/lib/localdb.js#L13-L19
This block of code https://github.com/feedhenry/fh-mbaas-api/blob/master/lib/db.js#L121-L123 where It looks for the `FH_USE_LOCAL_DB` env var (which is set in the `Gruntfile.js`) ensures it doesn't look for a ditch server locally (like there is in dynofarm), and instead tell it to connect directly to a mongdb.
However, the logic in our sync code is not going down the expected path for the mongdb version of the DataSet model when running locally. This is because the connection string check we have here https://github.com/feedhenry/fh-mbaas-api/pull/41/files#diff-b741a8a6262978a76c0f39c1b9fe2f78R76 doesn't consider the `FH_USE_LOCAL_DB` env var in this block https://github.com/feedhenry/fh-mbaas-api/blob/master/lib/db.js#L59-L70, and so it thinks we don't have a direct connection to mongo.

I believe its a local development bug in the getMongoConnectionUrl function. It should probalby short circuit if `FH_USE_LOCAL_DB` is set, and call the callback with a local mongo connection string.

However, the reason I wasn't seeing this was, I was starting my cloud app with this command:
```
FH_MONGODB_CONN_URL=mongodb://test:test@127.0.0.1:27017/FH_LOCAL node application.js 
```
*and* I had configured a `test` user with a password of `test` with `dbAdmin` role on the `FH_LOCAL` database.

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* BLOCKER [Add a semicolon at the end of this statement.](https://github.com/feedhenry/fh-mbaas-api/blob/30033a4af4087021e8edba17eebcbdef840952e5/lib/sync-UpdatesModel_mongo.js#L29) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

File: lib/db.js [hunk 2]
```diff
@@ -57,16 +57,21 @@ function mongoConnectionStringOS3(cb){
...
+    return cb(null, 'dummy');
+  }
   if(process.env.FH_MONGODB_CONN_URL){
     return cb(undefined, process.env.FH_MONGODB_CONN_URL);
-  } else if (process.env.OPENSHIFT_MONGODB_DB_HOST) {
+  }
+  if (process.env.OPENSHIFT_MONGODB_DB_HOST) {
     return mongoConnectionStringOS2(cb);
-  } else if ("openshift3" === process.env.FH_MBAAS_TYPE){
+  }
```
Comment (fh-qube) [inline lib/db.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* BLOCKER 1 blocker
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* BLOCKER [Add a semicolon at the end of this statement.](https://github.com/feedhenry/fh-mbaas-api/blob/d320a7167f2050cd6872a474f60b81559662d01a/lib/sync-UpdatesModel_mongo.js#L29) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* BLOCKER 1 blocker

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* BLOCKER [Add a semicolon at the end of this statement.](https://github.com/feedhenry/fh-mbaas-api/blob/26f878186c9b6d939efdf485cf02f5afddf2c899/lib/sync-UpdatesModel_mongo.js#L29) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)

**Comment by danbev** [comment]
> I believe its a local development bug in the getMongoConnectionUrl function. It should probalby short circuit if FH_USE_LOCAL_DB is set, and call the callback with a local mongo connection string.

Not sure if [this](https://github.com/feedhenry/fh-mbaas-api/pull/43/commits/d320a7167f2050cd6872a474f60b81559662d01a) is what you had in mind as it looks pretty bad and I'm finding this code very hard to follow and make sense of to be honest.

**Comment by aidenkeating** [comment]
retest this please
